src/main/java/software/amazon/encryption/s3/materials/RsaKeyring.java

```diff
@@ -156,11 +155,6 @@ private byte[] parsePseudoDataKey(DecryptionMaterials materials, byte[] pseudoDa
             System.arraycopy(pseudoDataKey, 1, dataKey, 0, dataKeyLengthBytes);
             System.arraycopy(pseudoDataKey, 1 + dataKeyLengthBytes, dataCipherName, 0, dataCipherNameLength);
 
-            byte[] expectedDataCipherName = materials.algorithmSuite().cipherName().getBytes(StandardCharsets.UTF_8);
-            if (!Arrays.equals(expectedDataCipherName, dataCipherName)) {
-                throw new S3EncryptionClientException("The data cipher does not match the data cipher used for encryption. The object may be altered or corrupted");
-            }
```

Comment:
I would like to authenticate all the data. So removing the authentication is not the preferred solution. @lucasmcdonald3  and I are looking at options.

Reply:
Should we be replicating https://github.com/aws/amazon-s3-encryption-client-java/blob/a13ef235fed1b4243b46a537690063da0f8f5ab3/src/main/java/software/amazon/encryption/s3/materials/AesKeyring.java#L143C1-L143C132 ?